specification/deploymentmanager/resource-manager/Microsoft.DeploymentManager/preview/2019-11-01-preview/deploymentmanager.json

```diff
@@ -0,0 +1,2465 @@
+          "type": "string",
+          "readOnly": true,
+          "description": "Error code string."
+        },
+        "message": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Descriptive error information."
+        },
+        "target": {
+          "type": "string",
+          "description": "Error target"
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CloudErrorBody"
+          },
+          "description": "More detailed error information."
+        }
+      }
+    },
+    "OperationsList": {
+      "description": "The operations response.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The list of supported operations",
+          "$ref": "#/definitions/Operation"
+        }
+      }
+    },
+    "Operation": {
+      "description": "Represents an operation that can be performed on the service.",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the operation."
+        },
+        "display": {
+          "description": "The display name of the operation.",
+          "$ref": "#/definitions/OperationDetail"
+        },
+        "origin": {
+          "type": "string",
+          "description": "The origin of the operation."
+        },
+        "properties": {
+          "description": "The properties of the operation.",
+          "type": "object",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "OperationDetail": {
+      "description": "The detail about an operation.",
+      "properties": {
+        "provider": {
+          "type": "string",
+          "description": "The name of the provider that supports the operation."
+        },
+        "resource": {
+          "type": "string",
+          "description": "The resource type on which this operation can be performed."
+        },
+        "operation": {
+          "type": "string",
+          "description": "The name of the operation."
+        },
+        "description": {
+          "type": "string",
+          "description": "The description of the operation."
+        }
+      }
+    },
+    "StepResource": {
+      "description": "The resource representation of a rollout step.",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "The properties that define the step.",
+          "$ref": "#/definitions/StepProperties"
+        }
+      },
+      "required": [
+        "properties"
+      ]
+    },
+    "StepProperties": {
+      "description": "The properties of a step resource.",
+      "type": "object",
+      "discriminator": "stepType",
+      "properties": {
+        "stepType": {
+          "type": "string"
```

Comment:
This should be an enum

Reply:
Added enum values block inside. Type remains as string, per swagger specification.